### ISSUE TITLE
Service: `no public ports are exposed`

### ISSUE THREAD

**Issue by 0cv** [sst/ion#795]
This is a recent issue, I've pinpointed the regression between version 0.1.42 where it works and 0.1.43 where it fails

Running `sst dev --stage dev --verbose`

`sst.config.ts`
```js
/// <reference path="./.sst/platform/config.d.ts" />

export default $config({
  app(input) {
    return {
      name: "abc",
      removal: "remove",
      home: "aws",
      providers: { aws: true },
      autostart: true,
    };
  },
  async run() {
    const vpc = new sst.aws.Vpc("Vpc");
    const cluster = new sst.aws.Cluster("Ecs", { vpc });
    cluster.addService("Service", {
      image: {
        context: "./service",
      },
      architecture: "arm64",
      cpu: "0.25 vCPU",
      memory: "0.5 GB",
    });

    return {
      cluster: cluster.nodes.cluster.arn,
    };
  },
});
```

<img width="395" alt="image" src="https://github.com/user-attachments/assets/f746adcc-05f1-4612-9095-c1e412133bd5">

EDIT:
This error between 0.1.42 and 0.1.43 happens ONLY when running `sst dev`. When running `sst deploy`, it *also* fails on 0.1.42. I've found that the regression for `sst deploy` happens between 0.1.40 where it works and 0.1.41 where it fails

**Comment by davearel** [comment]
I'm getting this as well

```ts
const vpc = new sst.aws.Vpc("Vpc")
const cluster = new sst.aws.Cluster("Cluster", {
  vpc,
})
export const api = cluster.addService("Api", {
  link: [...allSecrets, database],
  public: {
    ports: [{ listen: "80/http", forward: "3000/http" }],
  },
  dev: {
    command: "bun --watch src/index.ts",
  },
  image: {
    context: "platform/api",
  },
})
```
Dev command doesn't seem to be preventing it from deploying in dev, either?

**Comment by fwang** [comment]
@0cv @davearel Fix will be in the next release.

Please reopen this if issue persist.

**Comment by ericc59** [comment]
@fwang I'm still seeing this with v0.1.95

**Comment by bitcoinski** [comment]
Same in 0.1.95, "No public ports are exposed".

**Comment by anthonyringoet** [comment]
Can reproduce in `3.0.42`

**Comment by fwang** [comment]
Reopened.. looking into it.

**Comment by ryanleecode** [comment]
theres no error as long as u don't use `myService.url` in dev mode

**Comment by ryanleecode** [comment]
for those who aren't aware you can use `sst.aws.Vpc.v1` as a temporary workaround

**Comment by fwang** [comment]
Fixed in v3.0.73
